cmd/calyptia/delete_core_instance.go

```diff
@@ -0,0 +1,328 @@
+
+			cmd.Println("You confirm the deletion of those resources? [Y/n]")
+
+			confirmDelete = ask()
+
+			if !confirmDelete {
+				cmd.Println("operation canceled")
+				return nil
+			}
+
+			namespaces, err := k8sClient.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
+			if err != nil {
+				return err
+			}
+			for _, ns := range namespaces.Items {
+
+				err = k8sClient.DeleteDeploymentByLabel(ctx, label, ns.Name)
+				if err != nil {
+					if !skipError {
+						return err
+					}
+					fmt.Fprintf(cmd.OutOrStdout(), "a problem occurred while deleting deployments, err: %v\n", err)
+				}
+
+				services, err := k8sClient.FindServicesByLabel(ctx, label, ns.Name)
+				if err != nil {
+					return err
+				}
+
+				for _, item := range services.Items {
+					err := k8sClient.DeleteServiceByName(ctx, item.Name, ns.Name)
+					if err != nil {
+						if !skipError {
+							return err
+						}
+						fmt.Fprintf(cmd.OutOrStdout(), "a problem occurred while deleting %s, err: %v\n", item.Name, err)
+					}
+				}
+
+				serviceAcc := agg.Name + "-service-account"
+				err = k8sClient.DeleteServiceAccountByLabel(ctx, label, ns.Name)
+				if err != nil {
+					if !skipError {
+						return err
+					}
+					fmt.Fprintf(cmd.OutOrStdout(), "a problem occurred while deleting %s, err: %v\n", serviceAcc, err)
+				}
+
+				secret := agg.Name + "-private-rsa.key"
+				err = k8sClient.DeleteSecretByLabel(ctx, label, ns.Name)
+				if err != nil {
+					if !skipError {
+						return err
+					}
+					fmt.Fprintf(cmd.OutOrStdout(), "a problem occurred while deleting %s, err: %v\n", secret, err)
+				}
+			}
+			clusterRole := agg.Name + "-cluster-role"
+			err = k8sClient.DeleteClusterRoleByLabel(ctx, label)
+			if err != nil {
+				if !skipError {
+					return err
+				}
+				fmt.Fprintf(cmd.OutOrStdout(), "a problem occurred while deleting %s, err: %v\n", clusterRole, err)
+			}
+			roleBinding := agg.Name + "-cluster-role-binding"
+			err = k8sClient.DeleteRoleBindingByLabel(ctx, label)
+			if err != nil {
+				if !skipError {
+					return err
+				}
+				fmt.Fprintf(cmd.OutOrStdout(), "a problem occurred while deleting %s, err: %v\n", roleBinding, err)
+			}
+
+			err = config.cloud.DeleteAggregator(ctx, agg.ID)
+			if err != nil {
+				return err
+			}
+			fmt.Fprintf(cmd.OutOrStdout(), "calyptia-core instance %q was successfully deleted", agg.Name)
+			return nil
+		},
+	}
+	fs := cmd.Flags()
+	fs.BoolVar(&skipError, "skip-error", false, "Skip errors during delete process")
+	//isNonInteractiveMode := os.Stdin == nil || !term.IsTerminal(int(os.Stdin.Fd()))
+	//fs.BoolVar(&confirmDelete, "yes", isNonInteractiveMode, "Confirm deletion")
+	clientcmd.BindOverrideFlags(configOverrides, fs, clientcmd.RecommendedConfigOverrideFlags("kube-"))
+	return cmd
+}
+
+func listDeletionsByLabel(ctx context.Context, k8sClient *k8s.Client, cmd *cobra.Command, label string) error {
+	namespaces, err := k8sClient.CoreV1().Namespaces().List(ctx, metav1.ListOptions{})
```

Comment:
So, the user sets the namespace with the flag `--kube-namespace` but here it is using all namespaces, so effectively ignoring it.

Reply:
It doesn't matter, we'll iterate across all namespaces, including what is in kube-namespace provided by the user.